File: Examples/ObjectiveC/CustomCalloutViewExample.m
#import "CustomCalloutViewExample.h"
#import "CustomCalloutView.h"
@import Mapbox;

NSString *const MBXExampleCustomCalloutView = @"CustomCalloutViewExample";

@interface CustomCalloutViewExample () <MGLMapViewDelegate>
@end

@implementation CustomCalloutViewExample

- (void)viewDidLoad {
    [super viewDidLoad];

    MGLMapView *mapView = [[MGLMapView alloc] initWithFrame:self.view.bounds styleURL:[MGLStyle lightStyleURL]];
    mapView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    mapView.tintColor = [UIColor darkGrayColor];
    [self.view addSubview:mapView];

    // Set the map view‘s delegate property.
    mapView.delegate = self;

    // Initialize and add the marker annotation.
    MGLPointAnnotation *marker = [[MGLPointAnnotation alloc] init];
    marker.coordinate = CLLocationCoordinate2DMake(0, 0);
    marker.title = @"Hello world!";

    // This custom callout example does not implement subtitles.
    //marker.subtitle = @"Welcome to my marker";

    // Add the annotation to the map.
    [mapView addAnnotation:marker];

    // Select the annotation so the callout will appear.
    [mapView selectAnnotation:marker animated:NO completionHandler:nil];
}

- (BOOL)mapView:(MGLMapView *)mapView annotationCanShowCallout:(id <MGLAnnotation>)annotation {
    // Only show callouts for `Hello world!` annotation.
    return [annotation respondsToSelector:@selector(title)] && [annotation.title isEqualToString:@"Hello world!"];
}

- (UIView<MGLCalloutView> *)mapView:(__unused MGLMapView *)mapView calloutViewForAnnotation:(id<MGLAnnotation>)annotation
{
    // Instantiate and return our custom callout view.
    CustomCalloutView *calloutView = [[CustomCalloutView alloc] init];
    calloutView.representedObject = annotation;
    return calloutView;
}

- (void)mapView:(MGLMapView *)mapView tapOnCalloutForAnnotation:(id<MGLAnnotation>)annotation
{
    // Optionally handle taps on the callout.
    NSLog(@"Tapped the callout for: %@", annotation);

    // Hide the callout.
    [mapView deselectAnnotation:annotation animated:YES];
}

@end

File: Examples/Swift/CustomCalloutViewExample.swift
//import Mapbox
//
//@objc(CustomCalloutViewExample_Swift)
//
//class CustomCalloutViewExample_Swift: UIViewController, MGLMapViewDelegate {
//    override func viewDidLoad() {
//        super.viewDidLoad()
//        
//        let mapView = MGLMapView(frame: view.bounds, styleURL: MGLStyle.lightStyleURL())
//        mapView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
//        mapView.tintColor = .darkGray
//        view.addSubview(mapView)
//        
//        // Set the map view‘s delegate property.
//        mapView.delegate = self
//        
//        // Initialize and add the marker annotation.
//        let marker = MGLPointAnnotation()
//        marker.coordinate = CLLocationCoordinate2D(latitude: 0, longitude: 0)
//        marker.title = "Hello world!"
//        
//        // This custom callout example does not implement subtitles.
//        //marker.subtitle = "Welcome to my marker"
//        
//        // Add marker to the map.
//        mapView.addAnnotation(marker)
//
//        // Select the annotation so the callout will appear.
//        mapView.selectAnnotation(marker, animated: false)
//    }
//    
//    func mapView(_ mapView: MGLMapView, annotationCanShowCallout annotation: MGLAnnotation) -> Bool {
//        // Only show callouts for `Hello world!` annotation.
//        return annotation.responds(to: #selector(getter: MGLAnnotation.title)) && annotation.title! == "Hello world!"
//    }
//
//    func mapView(_ mapView: MGLMapView, calloutViewFor annotation: MGLAnnotation) -> MGLCalloutView? {
//        // Instantiate and return our custom callout view.
//        return CustomCalloutView(representedObject: annotation)
//    }
//    
//    func mapView(_ mapView: MGLMapView, tapOnCalloutFor annotation: MGLAnnotation) {
//        // Optionally handle taps on the callout.
//        print("Tapped the callout for: \(annotation)")
//        
//        // Hide the callout.
//        mapView.deselectAnnotation(annotation, animated: true)
//    }
//}

